### PR TITLE
Add support for windows build

### PR DESCRIPTION
**Pull request by CoderHam**

### CODE REVIEW

File: src/identity.cc [hunk 1]
```diff
@@ -338,7 +339,7 @@ extern "C" {
 // Implementing TRITONBACKEND_Initialize is optional. The backend
 // should initialize any global state that is intended to be shared
 // across all models and model instances that use the backend.
-TRITONSERVER_Error*
+TRITONBACKEND_ISPEC TRITONSERVER_Error*
```
Comment (deadeyegoodwin) [inline src/identity.cc:343]: You shouldn't need to put ISPEC yourself (where is it even defined?). It seems you are missing include of tritonbackend.h which will cause these to be dllexport.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/identity.cc (hunk 1 above)
Comment (CoderHam) [reply src/identity.cc:343]: Yes, I missed the tritonbackend header include.  I didn't think I needed to but once I did, it worked as expected. I also noticed that the [onnxruntime backend](https://github.com/triton-inference-server/onnxruntime_backend/blob/main/src/onnxruntime.cc) adds it manually as well.

Comment (deadeyegoodwin) [reply src/identity.cc:343]: Can you fix onnxruntime_backend as well.

Comment (CoderHam) [reply src/identity.cc:343]: Maybe I am misunderstanding, but why is TRITONBACKEND_ISPEC not required?

Comment (deadeyegoodwin) [reply src/identity.cc:343]: I would think that the ISPEC (dllexport) is only needed in the tritonbackend.h file on the declaration of the function. I think that it is not needed on the definition of the function. https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-declspec-dllexport?view=msvc-160
Note other backends, except for onnxruntime and openvino which incorrectly have the ispec on the definition.
What happens if you include tritonbackend.h and remove ISPEC from identity.cc.

Comment (CoderHam) [reply src/identity.cc:343]: Will try that and update here

Comment (CoderHam) [reply src/identity.cc:343]: When I build with your suggested changes the functions are not exported by the dll. 

```
dumpbin.exe /exports ..\install\backends\identity\triton_identity.dll
```

Is empty. While when I retain the ISPEC the functions are successfully exported by the dll.

Comment (deadeyegoodwin) [reply src/identity.cc:343]: Is tritonbackend.h being included? It must be given that is where ISPEC is defined. Perhaps windows requires dllexport on the definition? That doesn't seem right. Could you create a single example to test?

Comment (CoderHam) [reply src/identity.cc:343]: Yes, tritonbackend.h is being included. By a single example you mean adding ISPEC to only one of the functions?
I added the ISPEC just for one function declaration and only that function was exported according to dumpbin

Comment (CoderHam) [reply src/identity.cc:343]: TODO: Follow up why ISPEC is needed in declarations in each backend. This is ready to be merged as is

Comment (deadeyegoodwin) [reply src/identity.cc:343]: Not a priority...

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**